services/city_service_test.go

```diff
@@ -0,0 +1,369 @@
+package services
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"sync"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"go.uber.org/mock/gomock"
+	"gorm.io/gen"
+
+	"github.com/NikolNikolaeva/project_weather/config"
+	api "github.com/NikolNikolaeva/project_weather/generated/api/project-weather/rest"
+	"github.com/NikolNikolaeva/project_weather/generated/api/weatherapi"
+	"github.com/NikolNikolaeva/project_weather/generated/dao/model"
+	mock_repositories "github.com/NikolNikolaeva/project_weather/generated/go-mocks/repositories"
+	mock_resources "github.com/NikolNikolaeva/project_weather/generated/go-mocks/resources"
+	mock_services "github.com/NikolNikolaeva/project_weather/generated/go-mocks/services"
+)
+
+func TestAPIService_GetById_CityNotFound(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mockRepo := mock_repositories.NewMockCityRepo(ctrl)
+	mockConverter := mock_resources.NewMockConverterI(ctrl)
```

Comment:
Why do we need the converter to be mocked? The more mocks you use in your unit tests the more useless they become. Let's keep the mocks to a minimum and use as much real code as possible.